tests/common/include/tests/common/tools_navigator.inl

```diff
@@ -19,7 +18,16 @@
 
 namespace detray {
 
-/** Checks for a correct 'towards_surface' state */
+namespace {
+
+// dummy propagator state
+template <typename stepping_t, typename navigation_t>
+struct prop_state {
+    stepping_t &_stepping;
+    navigation_t &_navigation;
```

Comment:
Can we use normal value rather than reference?

Reply:
It is just a temporary definition when there is no propagator that could take ownership of these states. And it is quite a bit more to write since I have to retrieve the navigation and stepper state from the propagation state again. If I have a dummy navigator state, then I have to give a separate instance to every ```prop_state```. I changed it now, but I am not a big fan. I would rather roll it back again

Reply:
I tried to reduce that a little by also changing the ```navigator``` interface to take a propagation state now. While I was at it, I also updated the navigator comments and added a few small fixes.